release notes/v0.39.0.md

```diff
@@ -0,0 +1,60 @@
+
+# Roadmap and future plans
+
+As the lack of big changes above suggests, over the last few months we have focused our efforts in a few areas that haven't yet been merged into the core of k6.
+
+We are introducing this new section in the release notes to inform the community of important features that we are currently working on - our short-term roadmap in a sense. We will also use it to give notice of breaking changes we plan to make in the near future.
+
+## `k6/experimental/*` JS modules
+
+Over the last several k6 releases, among a lot of other refactoring, we have added support for JavaScript event loops ([#2228](https://github.com/grafana/k6/pull/2228)) in k6 VUs and added a new Go API for exposing built-in and xk6 extension modules to user scripts ([announcement](https://github.com/grafana/k6/blob/master/release%20notes/v0.36.0.md#extensions), [docs](https://k6.io/docs/extensions/guides/create-an-extension/#advanced-javascript-extension)). This has given us (and any xk6 extension authors!) the ability to better support various asynchronous streaming/messaging/etc. protocols ([#882](https://github.com/grafana/k6/issues/882#issue-392218655)).
+
+We have started building some of these newly possible APIs as [xk6 extensions](https://k6.io/docs/extensions/) first, to be able to iterate on them more quickly and get user some feedback while we are building them. [xk6-websockets](https://github.com/grafana/xk6-websockets), [xk6-timers](https://github.com/grafana/xk6-timers) and [xk6-redis](https://github.com/grafana/xk6-redis/) are some of the first such APIs, but we plan to also work on support for gRPC streaming ([#2020](https://github.com/grafana/k6/issues/2020)), messaging protocols ([#1269](https://github.com/grafana/k6/issues/1269)), a new and better HTTP API ([#2461](https://github.com/grafana/k6/issues/2461)) and many others in the future!
+
+We want to eventually include a lot of these APIs into the k6 core, as built-in modules that users can directly use, without the need to mess with xk6 or Go compilation. However, because we try to keep the built-in k6 APIs stable and backwards-compatible, we want to get more user feedback before we do that, while we are still free to iterate and make (hopefully minor) breaking changes.
+
+So, we decided to create a new middle ground between the unstable and purely external xk6 extensions and the stable built-in k6 APIs - built-in `k6/experimental/*` modules! Our goal is that, starting with the next k6 v0.40.0 release, we'll start releasing some or all of these core-bound extensions as built-in k6 modules under these `k6/experimental/` import paths. This will allow all k6 users, both OSS and Cloud, to give us feedback and help us improve the APIs before we stabilize them.
```

Comment:
```suggestion
So, we decided to create a new middle ground between the unstable and purely external xk6 extensions and the stable built-in k6 APIs―built-in `k6/experimental/*` modules! Our goal is that, starting with the next k6 v0.40.0 release, we'll start releasing some or all of these core-bound extensions as built-in k6 modules under these `k6/experimental/` import paths. This will let k6 users, both OSS and Cloud, to give us feedback and help us improve the APIs before we stabilize them.
```